### ISSUE TITLE
[Request] "Remove all" options for machines, or access to recipe representations

### ISSUE THREAD

**Issue by nihiltres** [Shadows-of-Fire/EnderTweaker#13]
An option to strip every recipe from each machine (individually, e.g. `AlloySmelter.removeAllRecipes()`) would be very valuable. I'm trying to disable most of the "main" Ender IO machines (to add the "soul" machines and such but "not add" the SAG Mill etc.) and for cleanliness' sake I'm removing every recipe from each disabled machine.

If this option is undesirable, another possibility might be to give access to the list of recipes such that scripters could do stuff like this:

    for asr in AlloySmelter.recipes {
        if (testArbitraryCondition(asr)) {asr.remove();}
        if (otherTest(asr)) {
            asr.remove();
            AlloySmelter.addRecipe(asr.output, asr.input, asr.energyCost / 2, asr.xp * 2);
        }
    }
Of course, while the latter is more powerful, it seems probably more work to implement.

**Comment by Shadows-of-Fire** [comment]
If the machines are disabled it would likely be easier to hide the jei category and forget about it.

Anyhow, its unlikely I work on anything for 1.12 anymore, but you could probably accomplish removing all recipes by looping over the item registry and calling remove a bunch

**Comment by VariableVixen** [comment]
Nope, or at least not well - any `AlloySmelter.remove(IItemStack)` call for an item that doesn't have an Alloy Smelter recipe produces an error not just in the log file, but also in the game chat. Going over the item registry and calling `.remove()` on everything is Not A Good Idea.
